source/jormungandr/jormungandr/realtime_schedule/synthese.py

```diff
@@ -0,0 +1,176 @@
+# coding=utf-8
+
+# Copyright (c) 2001-2016, Canal TP and/or its affiliates. All rights reserved.
+#
+# This file is part of Navitia,
+#     the software to build cool stuff with public transport.
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Canal TP (www.canaltp.fr).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+from __future__ import absolute_import, print_function, unicode_literals, division
+
+from jormungandr.realtime_schedule.realtime_proxy import RealtimeProxy
+from jormungandr.schedule import RealTimePassage
+import xml.etree.ElementTree as et
+from jormungandr.interfaces.parsers import date_time_format
+import pytz
+from flask import logging
+import pybreaker
+import requests as requests
+from jormungandr import cache, app
+from datetime import datetime, time
+
+class SyntheseRoutePoint(object):
+
+    def __init__(self, rt_id=None, sp_id=None):
+        self.syn_route_id = rt_id
+        self.syn_stop_point_id = sp_id
+
+    def __key(self):
+        return (self.syn_route_id, self.syn_stop_point_id)
+
+    def __hash__(self):
+        return hash(self.__key())
+
+    def __eq__(self, other):
+        return self.__key() == other.__key()
+
+    def __repr__(self):
+        return "SyntheseRoutePoint({})".format(self.__key())
+
+
+class Synthese(RealtimeProxy):
+    """
+    class managing calls to timeo external service providing real-time next passages
+    """
+
+    def __init__(self, id, service_url, timezone, timeout=10):
```

Comment:
:sob: this class will be broken with my PR: https://github.com/CanalTP/navitia/pull/1506